dotnet/samples/Demos/AgentWithHistoryTruncation/TruncatingChatAgent.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.Agents;
+
+namespace AgentWithHistoryTruncation;
+
+internal class TruncatingChatAgent : ChatCompletionAgent
+{
+    /// <summary>
+    /// %%%
+    /// </summary>
+    public TruncationStrategy? TruncationStrategy { get; init; }
```

Comment:
Why not just put this on `ChatHistoryKernelAgent` and iterate with partner on the trunction-strategy and just go _all the way_?